### PR TITLE
TF2 quickstart

### PR DESCRIPTION
**Pull request by Le-Zheng**
Colab notebook link:  https://colab.research.google.com/github/Le-Zheng/analytics-zoo/blob/tf2doc/docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb

For https://github.com/intel-analytics/analytics-zoo/issues/3372

### CODE REVIEW

File: readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    init_orca_context(cluster_mode="yarn-client", num_nodes=2, cores=2, init_ray_on_spark=True) # run on Hadoop YARN cluster
+```
+
+This is the only place where you need to specify local or distributed mode. View [Orca Context](./../Overview/orca-context.md) for more details.
+
+**Note:** You should `export HADOOP_CONF_DIR=/path/to/hadoop/conf/dir` when running on Hadoop YARN cluster. View [Hadoop User Guide](./../../UserGuide/hadoop.md) for more details.
+
+### **Step 2: Define the Model**
+
+You may define your model, loss and metrics in the same way as in any standard (single node) TensorFlow program.
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md:36]: You can then define your model, loss and metrics in _Creator Functions_ using the standard TensroFlow 2 APIs.

File: readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    )
+
+    model.compile(optimizer=tf.keras.optimizers.RMSprop(),
+                  loss='sparse_categorical_crossentropy',
+                  metrics=['accuracy'])
+    return model
+```
+### **Step 3: Define Train Dataset**
+
+You can define the dataset using standard [tf.data.Dataset](https://www.tensorflow.org/api_docs/python/tf/data/Dataset). Orca also supports [Spark DataFrame](https://spark.apache.org/docs/latest/sql-programming-guide.html) and [Orca XShards](./data).
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md:62]: You can define the dataset in the _Creator Function_ using standard [tf.data.Dataset](https://www.tensorflow.org/api_docs/python/tf/data/Dataset) APIs.

File: readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                steps_per_epoch=60000 // batch_size,
+                validation_data=val_data_creator,
+                validation_steps=10000 // batch_size)
+                
+est.save("/tmp/mnist_keras.ckpt")
+est.load("/tmp/mnist_keras.ckpt")
+
+stats = est.evaluate(val_data_creator, num_steps=10000 // batch_size)
+print(stats)
+```
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md:113]: need to call est.shutdown

File: docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+        "id": "QAQp0FcUWaH3"
+      },
+      "source": [
+        "# recommended to set it to True when running Analytics Zoo in Jupyter notebook \r\n",
+        "OrcaContext.log_output = True # (this will display terminal's stdout and stderr in the Jupyter notebook).\r\n",
+        "\r\n",
+        "cluster_mode = \"local\"\r\n",
+        "\r\n",
+        "if cluster_mode == \"local\":  \r\n",
+        "    init_orca_context(cluster_mode=\"local\", cores=4, init_ray_on_spark=True) # run in local mode\r\n",
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb:190]: no need to specify `init_ray_on_spark`

File: readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+conda create -n zoo python=3.7 # "zoo" is conda environment name, you can use any name you like.
+conda activate zoo
+pip install analytics-zoo[ray] # install either version 0.9 or latest nightly build
+pip install tensorflow==2.3.0
+```
+
+### **Step 1: Init Orca Context**
+```python
+if args.cluster_mode == "local":
+    init_orca_context(cluster_mode="local", cores=4, init_ray_on_spark=True)# run in local mode
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md:23]: No need to specify `init_ray_on_spark`

File: readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+Next, fit and evaluate using the Estimator.
+```python
+stats = est.fit(train_data_creator,
+                epochs=5,
+                steps_per_epoch=60000 // batch_size,
+                validation_data=val_data_creator,
+                validation_steps=10000 // batch_size)
+                
+est.save("/tmp/mnist_keras.ckpt")
+est.load("/tmp/mnist_keras.ckpt")
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md:109]: move save to the end, and no need to show load

File: docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "uWQaFJBf55LJ"
+      },
+      "source": [
+        "# Install required dependencies\r\n",
+        "!pip install tensorflow==2.3.0 "
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb:138]: do we need to install TF 2.3 here?

File: docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "imhbbDNXXsXR"
+      },
+      "source": [
+        "max_epoch=5\r\n",
+        "stats = est.fit(train_data_creator,\r\n",
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb:334]: this seems to be much faster than TF 1.15?

File: docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "yyRtP6lqieej"
+      },
+      "source": [
+        "Now, the accuracy of this model has reached 98%."
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb:372]: I only see 96% accuracy?

File: docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb (hunk 7 above)
Comment (Le-Zheng) [reply docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb]: Sure, we don't need.

File: docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb (hunk 8 above)
Comment (Le-Zheng) [reply docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb:324]: Yes, it takes only 30s to train 5 epochs.

File: docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb (hunk 9 above)
Comment (Le-Zheng) [reply docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb]: 96%-97%

File: readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md (hunk 3 above)
Comment (Le-Zheng) [reply readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md:113]: Added.

File: readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+---
+
+**In this guide we will describe how to scale out TensorFlow (v2.3) programs using Orca in 4 simple steps.**
+
+<a target="_blank" href="https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb"><img src="https://www.tensorflow.org/images/colab_logo_32px.png" />Run in Google Colab</a>&nbsp; <a target="_blank" href="https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb"><img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png" />View source on GitHub</a>
+
+### **Step 0: Prepare Environment**
+
+We recommend using [Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) to prepare the environment. Please refer to the [install guide](../../UserGuide/python.md) for more details.
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md:11]: change to `conda`, not `Conda`

File: readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
+# TensorFlow 2.3 Quickstart
+
+---
+
+**In this guide we will describe how to scale out TensorFlow (v2.3) programs using Orca in 4 simple steps.**
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-tf2-quickstart.md:5]: to scale out TensorFlow 2 program

File: docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb [hunk 12]
```diff
@@ -0,0 +1,378 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "l-l4vel5N3qP"
+      },
+      "source": [
+        "## **Distributed TensorFlow (v2.3) using Orca APIs**\r\n",
+        "\r\n",
+        "In this guide we will describe how to scale out TensorFlow (v2.3) programs using Orca in 4 simple steps."
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb:139]: how to scale out TensorFlow 2 programs

File: docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb [hunk 13]
```diff
@@ -0,0 +1,378 @@
...
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "l-l4vel5N3qP"
+      },
+      "source": [
+        "## **Distributed TensorFlow (v2.3) using Orca APIs**\r\n",
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb:137]: Distributed TensorFlow 2 using Orca APIs

**Comment by Le-Zheng** [comment]
Jenkins passed: http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Colab-TF2/1/console

**Comment by jason-dai** [comment]
LGTM
